pkg/devfile/parser/parse_test.go

```diff
@@ -133,6 +135,27 @@ func Test_parseParentAndPluginFromURI(t *testing.T) {
 									},
 								},
 							},
+							{
+								Name: "image",
```

Comment:
those image components are basically identical, except for those boolean values. Can you create a function to generate the image component and accept the boolean value as a param? and call that func directly instead of re-defining it in each test

Reply:
good point, optimizations can be made in other components too.  I'll try to get those updated as well.

Reply:
There are some common functions for other components in: https://github.com/devfile/library/blob/main/pkg/testingutil/devfile.go  , you can add image component in that file as well

for those tests that need more flexibility on component content can stay with the definition in unit test.

Reply:
I created some utils and I was able to extend it so other properties could be set.  If you agree with this, I think we should look  at cleaning up some areas of our test cases in a similar manner since it produces a more readable file